contents/security/passwordStorage.md

```diff
@@ -0,0 +1,106 @@
+
+Thus, the way to store passwords properly is to use a salted hash - take the user's password, append some bytes to it, and hash the result. That hash is the user's password. When the user attempts to login again, perform the same procedure again. If the hash matches, then you know that the user is who he says he is, even if you don't actually know the original password that the user provided. 
+
+One question that is commonly asked by developers is where to store the salt. The salt can be stored in plaintext, along with the user in the database. Since the goal of the salt is only to prevent precomputed rainbow tables from being used, it doesn't need to be encrypted in the database.
+
+### If an attacker has gained access to the entire application, what does it matter if my passwords are stored in plaintext or not?
+
+If an attacker has already gained access to the entire application, then he already has all the information hat he coculd possibly get. He would have access to all of your application's data, including data from your users or from any analytics software that you might be running. However, by adding salt and hashing your passwords, the attacker still doesn't know your customer's passwords and could take years to find out. Otherwise, since 59% of people use the same password across multiple sites<sup>1</sup>, the attacker could quickly try other websites such as banks to attempt to break into those accounts, which can yield great financial return.
+
+Additionally, when a user signed up on your website and provided you a password, they implicitly trusted you to keep that information safe and secure for them, and in a sense you have a responsibility to keep their passwords secure as well. By doing proper password storage, if your servers ever get breached, you can assure your customers that their passwords are properly secured and maintain some of their trust in you.
+
+## Word of warning
+
+Reading this article doesn't make you an expert in writing your own password hashing code. It is far too easy to screw up and make a mistake. Instead, use one of the free libraries that provide a crypto function that has already been well-tested by the community. **Do not write your own crypto library.** For example, here are some libraries you can use instead depending on your language:
```

Comment:
Personally, I feel that this comes off a bit strong and aggressive (but, I guess that's the point) 😅 

You can put this into a `Getting Started` section instead

It may be better to just recommend these to the reader instead of telling them to not try to do it themselves. At the end of the day, these libraries still have to be made by someone

Reply:
I do feel that it might be appropriate in this case though, since one of the most common advice (and for good reason) regarding security is "don't roll your own crypto". 😅 

Do you feel like putting this in a warning box might make this point stand out more to readers?